src/clj/battlebots/arena/generation.clj

```diff
@@ -37,31 +38,86 @@
   [amount item arena]
   (reduce replacer arena (repeat amount item)))
 
+(defn- wrap
+  "Wraps out-of-bounds coordinates (zero-based) to opposite edge of m x n arena"
+  [[x y] [m n]]
+  {:pre [(integer? x) (integer? y) (pos? m) (pos? n)]
```

Comment:
Yep.  And they have access to destructured args *and* can be set per-signature for polymorphic (by arg count) functions.